gui/src/components/NumberView.tsx

```diff
@@ -0,0 +1,85 @@
+import { IconButton, Menu, MenuItem } from "@mui/material";
+import { MoreVert as MoreVertIcon } from "@mui/icons-material";
+import { useState } from "react";
+import { formatEther, formatGwei, parseEther } from "viem";
+
+import { Typography } from "@ethui/react/components";
+import { ContextMenuWithTauri } from "./ContextMenuWithTauri";
+import { useNetworks } from "@/store";
+
+interface Props {
+  value: number;
+}
+
+export function NumberView({ value }: Props) {
+  const [displayValue, setDisplayValue] = useState(value.toString());
+  const [menuAnchor, setMenuAnchor] = useState<null | HTMLElement>(null);
+  const [formatValue, setFormatValue] = useState("Hexadecimal");
+
+  const network = useNetworks((s) => s.current);
+  if (!network) return null;
+
+  const formatGweiValue = () => {
+    setDisplayValue(formatGwei(BigInt(value)) + " gwei");
+    setMenuAnchor(null);
+  };
+
+  const formatWeiValue = () => {
+    const weiValue = parseEther(formatEther(BigInt(value)).toString());
+    setDisplayValue(weiValue.toString() + " wei");
+    setMenuAnchor(null);
+  };
+
+  const formatEtherValue = () => {
+    setDisplayValue(formatEther(BigInt(value)) + " eth");
+    setMenuAnchor(null);
+  };
+
+  const changeFormat = () => {
+    if (formatValue == "Decimal") {
+      setFormatValue("Hexadecimal");
+      setDisplayValue(value.toString());
+    } else if (formatValue == "Hexadecimal") {
+      setFormatValue("Decimal");
+      const numberValue = Number(value);
+      const hexadecimalValue = numberValue.toString(16).toUpperCase();
+      setDisplayValue("0x" + hexadecimalValue);
+    } else {
+      setDisplayValue(value.toString());
+    }
+    setMenuAnchor(null);
+  };
```

Comment:
a "Change Format" button on the UI is very unclear as to what its purpose is.
from seeing the code, seems to toggle between hex and decimal notation. but for a user that won't be clear at all